llvm/lib/Target/AMDGPU/AMDGPUInstructionSelector.cpp

```diff
@@ -639,7 +639,7 @@ bool AMDGPUInstructionSelector::selectG_MERGE_VALUES(MachineInstr &MI) const {
   LLT SrcTy = MRI->getType(MI.getOperand(1).getReg());
 
   const unsigned SrcSize = SrcTy.getSizeInBits();
-  if (SrcSize < 32)
+  if (SrcSize < 16)
```

Comment:
Why are these two cases in the same patch?

Reply:
Since the the ICMP one has doubt, I'll move this to a seperate PR

Reply:
https://github.com/llvm/llvm-project/pull/129287